### PR TITLE
Replace 0/NULL with nullptr

### PR DESCRIPTION
**Pull request by economon**
## Proposed Changes
*Give a brief overview of your contribution here in a few sentences.*
 
Also removes checks for nullptr before delete. These checks are not necessary, since deleting nullptr is perfectly valid.

## Related Work
*Resolve any issues (bug fix or feature request), note any related PRs, or mention interactions with the work of others, if any.*

N/A

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [x] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags).
- [x] My contribution is commented and consistent with SU2 style.

### CODE REVIEW

File: SU2_CFD/src/drivers/CDriver.cpp [hunk 1]
```diff
@@ -501,34 +501,34 @@ void CDriver::Postprocessing() {
...
         delete output_container[iZone];
       }
     }
     delete [] output_container;
   }
 
-  if(driver_output != NULL){
+  
     delete driver_output;
-  }
```
Comment (pcarruscag) [inline SU2_CFD/src/drivers/CDriver.cpp:531]: alignment

File: SU2_CFD/src/drivers/CDriver.cpp [hunk 2]
```diff
@@ -3404,8 +3404,8 @@ CHBDriver::~CHBDriver(void) {
   unsigned short kInst;
 
   /*--- delete dynamic memory for the Harmonic Balance operator ---*/
-  for (kInst = 0; kInst < nInstHB; kInst++) if (D[kInst] != NULL) delete [] D[kInst];
-  if (D[kInst] != NULL) delete [] D;
+  for (kInst = 0; kInst < nInstHB; kInst++) if (D[kInst] != nullptr) delete [] D[kInst];
+  if (D[kInst] != nullptr) delete [] D;
```
Comment (pcarruscag) [inline SU2_CFD/src/drivers/CDriver.cpp:3408]: (not your code) but this does not look right.

File: SU2_CFD/src/fluid_model.cpp [hunk 3]
```diff
@@ -47,14 +47,14 @@ CFluidModel::CFluidModel(void) {
...
+  LaminarViscosity = nullptr;
+  ThermalConductivity = nullptr;
 
 }
 
 CFluidModel::~CFluidModel(void) {
-  if (LaminarViscosity!= NULL) delete LaminarViscosity;
-  if (ThermalConductivity!= NULL) delete ThermalConductivity;
+  delete LaminarViscosity;
+  delete ThermalConductivity;
```
Comment (pcarruscag) [inline SU2_CFD/src/fluid_model.cpp:57]: Glad to see the null checks going away.

File: SU2_CFD/src/interfaces/cht/CConjugateHeatInterface.cpp [hunk 4]
```diff
@@ -189,10 +189,10 @@ void CConjugateHeatInterface::GetDonor_Variable(CSolver *donor_solution, CGeomet
     Donor_Variable[3] = Tnormal*donor_config->GetTemperature_Ref();
   }
 
-  if (Edge_Vector != NULL) {
+  
 
     delete [] Edge_Vector;
-  }
```
Comment (pcarruscag) [inline SU2_CFD/src/interfaces/cht/CConjugateHeatInterface.cpp:195]: more alignment

File: SU2_CFD/src/numerics/elasticity/CFEANonlinearElasticity.cpp [hunk 5]
```diff
@@ -197,22 +197,22 @@ CFEANonlinearElasticity::~CFEANonlinearElasticity(void) {
...
+      if (b_Mat_Iso[iVar] != nullptr) delete [] b_Mat_Iso[iVar];
     }
     delete [] b_Mat_Iso;
   }
 
-  if (EField_Ref_Unit  != NULL)   delete [] EField_Ref_Unit;
-  if (EField_Ref_Mod   != NULL)   delete [] EField_Ref_Mod;
-  if (EField_Curr_Unit != NULL)   delete [] EField_Curr_Unit;
+    delete [] EField_Ref_Unit;
+    delete [] EField_Ref_Mod;
```
Comment (pcarruscag) [inline SU2_CFD/src/numerics/elasticity/CFEANonlinearElasticity.cpp:214]: more alignment, I'm guessing you have a script for this that can be fine tuned?

File: SU2_CFD/src/output/COutput.cpp [hunk 6]
```diff
@@ -180,12 +180,12 @@ COutput::~COutput(void) {
...
 
-  if (volumeDataSorter != nullptr)
+  
     delete volumeDataSorter;
 
   volumeDataSorter = nullptr;
 
-  if (surfaceDataSorter != nullptr)
+  
     delete surfaceDataSorter;
```
Comment (pcarruscag) [inline SU2_CFD/src/output/COutput.cpp:189]: more alignment.

**Comment by economon** [comment]
Fixed up the alignment issues, but I really think it is time that we move to an auto-formatter as a project (clang-format or another tool). It is not sustainable in a large project such as this to manually manage it.

We can do it as a pre-commit hook, which is a bit harsh as it forces everyone to install the formatter and could cause issues at commit time for folks, or we could also do it as a GitHub action perhaps. For instance, ninja can already do this for us as a one liner: install clang-format, drop a .clang-format file into the root SU2/ directory, configure a build, cd into build folder and ```$ ninja clang-format``` = entire project auto-formatted into a style of our choosing.

**Comment by talbring** [comment]
I like the idea of doing the formatting with ninja! We just have to decide on a format. Maybe someone can prepare a base clang-format file and then we discuss?

**Comment by economon** [comment]
> I like the idea of doing the formatting with ninja! We just have to decide on a format. Maybe someone can prepare a base clang-format file and then we discuss?

That's easy :). Try this in your .clang-format:

```
BasedOnStyle: Google
PointerAlignment: Left
DerivePointerAlignment: false
```

**Comment by pcarruscag** [comment]
Thanks for fixing the alignment.
With those options the code actually looks ok in areas that were recently cleaned up.
But the need for the manual formatting we have in loads of places comes from having code that is far too verbose and repetitive e.g. 10 calls to MPI routines with tons of arguments, 10 operator [] in a single line to get to the CSolver we need, entire copy pasted sections that should be functions, etc..
I am not a fan of that kind of wholesale automatic change because it does not fix any of the structural problems we have, and it tends to make the fixing of those issues harder because the aforementioned problematic areas are usually less readable after such changes.
I like the optimization approach of making small gradual improvements, but unfortunately this kind of maintenance is not very glamorous and few of us are up for it...

**Comment by economon** [comment]
I understand.. some of the lines are going to look very ugly after wrapping to 80 column widths, etc, but it will also highlight these trouble spots.

I see it as two orthogonal issues: (1) fix the formatting problems once and for all and have them automatically fixed in the future so we stop wasting time on it, and (2) we continue to fix the problem areas in the code structure as we go, which we will need to do anyway, regardless.

For me, I think the benefits of (1) outweigh the negative of seeing some ugly code.

**Comment by economon** [comment]
I have placed a .clang-format file in the root of the repo for anyone to use with clang-format. I recommend folks play around with it and apply it as needed on the files you change in future PRs. Eventually, we will enforce its use through a pre-commit hook or GitHub action.

p.s. bonus for VS Code users, this .clang-format will be used by default when using the auto-format shortcut (shift+option+F on Mac). Might check whether this is the case for other IDEs.

**Pull request merged**